gnmi/reconciler/reconciler.go

```diff
@@ -41,3 +45,124 @@ type Reconciler interface {
 	// ValidationPaths returns the set of path prefixes that the reconciler can validate.
 	ValidationPaths() []ygnmi.PathStruct
 }
+
+// Builder simplifies the creation of reconcilers and reduces some of the required boilerplate.
+type Builder struct {
+	br *builtReconciler
+}
+
+// Build returns the reconciler as configuration and resets the builder.
+func (b *Builder) Build() Reconciler {
+	reconciler := b.br
+	b.br = &builtReconciler{}
+	return reconciler
+}
+
+// WithID sets the id of the reconciler.
+func (b *Builder) WithID(id string) *Builder {
+	if b.br == nil {
+		b.br = &builtReconciler{}
+	}
+	b.br.id = id
+	return b
+}
+
+// WithStart appends a new start func to the reconciler.
+func (b *Builder) WithStart(startFn func(context.Context, *ygnmi.Client) error) *Builder {
+	if b.br == nil {
+		b.br = &builtReconciler{}
+	}
+	b.br.startFns = append(b.br.startFns, func(ctx context.Context, client gpb.GNMIClient, target string) error {
+		c, err := ygnmi.NewClient(client, ygnmi.WithTarget(target))
+		if err != nil {
+			return fmt.Errorf("failed to build client: %v", err)
+		}
+		return startFn(ctx, c)
+	})
+	return b
+}
+
+// WithValidator appends a validator and validations paths to the reconciler.
+func (b *Builder) WithValidator(paths []ygnmi.PathStruct, validator func(*oc.Root) error) *Builder {
+	if b.br == nil {
+		b.br = &builtReconciler{}
+	}
+	b.br.validateFns = append(b.br.validateFns, validator)
+	b.br.validationPaths = append(b.br.validationPaths, paths...)
```

Comment:
I'm wondering what the utility of storing this is. There is no way of checking that the user "only validated these paths and not others", right? So this is just for documentation purposes? What's wrong with just providing the reconciler with the whole config and let it validate away?

Reply:
see https://github.com/openconfig/lemming/pull/59#discussion_r983917638

it  a reconciler returns no paths, never need to run its validation func

Reply:
I see, ok, could you add a comment to the interface/implementation method? Looks like I've already forgotten.

Reply:
done